driver-core/src/test/java/com/datastax/driver/core/PagingStateTest.java

```diff
@@ -0,0 +1,109 @@
+package com.datastax.driver.core;
+
+import java.nio.ByteBuffer;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Iterator;
+
+import org.testng.annotations.Test;
+
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertNull;
+
+import com.datastax.driver.core.exceptions.InvalidPagingStateException;
+
+
+public class PagingStateTest extends CCMBridge.PerClassSingleNodeCluster {
+
+    @Override
+    protected Collection<String> getTableDefinitions() {
+        return Collections.singletonList("CREATE TABLE test (k text, v int, PRIMARY KEY (k, v))");
+    }
+
+    @Test(groups = "short")
+    public void changeInPagingStateFailsVerification (){
+
+        String key = "paging_test";
+        for (int i = 0; i < 100; i++)
+            session.execute(String.format("INSERT INTO test (k, v) VALUES ('%s', %d)", key, i));
+
+        SimpleStatement st = new SimpleStatement(String.format("SELECT v FROM test WHERE k='%s'", key));
+        ResultSet result = session.execute(st.setFetchSize(20));
+
+        Iterator<Row> rowIt = result.iterator();
+        for (int i = 0; i < 30; i++) {
+            int value = rowIt.next().getInt("v");
+        }
+        PagingState savedPagingState = result.getExecutionInfo().getPagingState();
+        String savedPagingStateString = savedPagingState.toString();
+
+        // corrupting the paging state
+        savedPagingState.getRawState().putInt(42);
+        ByteBuffer pagingstate = savedPagingState.getRawState().duplicate();
+
+        st = new SimpleStatement(String.format("SELECT v FROM test WHERE k='%s'", key));
+        try {
+            st.setFetchSize(20).setPagingState(savedPagingState);
+        }
+        catch (InvalidPagingStateException e) {
+            System.out.println(e.getMessage());
+        }
+        finally{
+            assertNull(st.getPagingState());
```

Comment:
OK this tests that setting the state failed, but it's more of a side-effect. You could have a bug where the state is not set but no exception is thrown, so I still think it's valuable to test for the exception.